### DISCUSSION TITLE
Dynamic Shift

### DISCUSSION THREAD

**Discussion by EricDallAgnol** [atoti/atoti#725] [Q&A] [answered]
Hi team 👋 

Currently, when it comes to implement shift, they are two functions exposed : 
- tt.function.shift
- tt.function.date_shift _(for LOCAL_DATE)_

However, the offset provided is always here a 'static' value. 
Is there a way to handle that dynamically ?

I'm doing a POC and I would like, through a parameter hierarchy to be able to shift on the fly in my measure 

Let's say I have a VaR 99% measure, my current date and the date with which I want to compare my VaR _(this second date value comes from a parameter hierarchy in my example)_ : 
![image](https://user-images.githubusercontent.com/83015366/205711066-a2fd8ffc-2558-482e-822d-2ad90611e3ab.png)

In this example I have a shift of -1 day thus I would like my VaR 99% Shifted measure to be shifted by one.
I'm wondering if there is a way to provide the offset based on my date_diff measure, so that if I select the date to be compared to another and my date_diff becomes, let's say -5 this time, then I would like my VaR 99% Shifted measure to be shifted by -5 (and to have only **ONE** VaR 99% shifted measure, not one shifted by -1 and another by -5)

Many thanks,

Eric

**Comment by tibdex** [comment, marked as answer]
Hi Eric,

Indeed, passing measures to `date_shift()` or `shift()`'s _offset_ parameter is not supported. As a workaround, if you have a small set of known in advance different offsets (let's say around 10), you could do something like that:

```py
tt.switch(
  m["date_diff"],
  {
    -1: tt.date_shift(m["VaR"], h["RiskDate"], offset="P-1D"),
    0: m["VaR"],
    ...,
    5: tt.date_shift(m["VaR"], h["RiskDate"], offset="P5D")
  }
)
```

You could even create the `switch()` _cases_ with a dict comprehension to have less things to type.

**Reply by EricDallAgnol** [reply]
Hi Thibault,

Thanks for your answer. Indeed the `switch()` can do the job - in my case it should be acceptable to handle that within predefined ranges.

Out of curiosity, would passing measures to `date_shift()` or `shift()` is on the medium/long term roadmap ? It would be a nice way to use the `date_diff()` function in my opinion.

Thanks !

Eric
